Source/JSONAPILinks.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  JSONAPILinks.swift
+//  Kakapo
+//
+//  Created by Joan Romano on 19/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+public enum JSONAPILink: CustomSerializable {
+    case Simple(value: String)
+    case Object(href: String, meta: Serializable)
+    
+    public func customSerialize() -> AnyObject? {
+        switch self {
+        case let Object(href, meta):
+            var serializedObject = [String: AnyObject](dictionaryLiteral: ("href", href))
+            serializedObject["meta"] = meta.serialize()
+            
+            return serializedObject
+        case let Simple(value):
+            return value
+        }
+    }
+}
+
+public protocol JSONAPILinkedEntity {
+    var links: [String : JSONAPILink]? { get }
+    var topLinks: [String : JSONAPILink]? { get }
```

Comment:
Neither me, but we need a way to differentiate between:

- Links in objects: inside `data` and inside `included` objects
- The very top `links` and links inside `relationships` (which is always one regardless of being an array or not)